pallets/author-inherent/src/lib.rs

```diff
@@ -106,7 +110,20 @@ pub mod pallet {
 				<Author<T>>::put(&author);
 			}
 
-			T::DbWeight::get().writes(1)
+			InherentIncluded::<T>::put(false);
+
+			// on_initialize: 2 writes
+			// on_finalize: 1 read
+			T::DbWeight::get().reads_writes(1, 2)
+		}
+		fn on_finalize(_: BlockNumberFor<T>) {
+			// According to parity, the only way to ensure that a mandatory inherent is included
+			// is by checking on block finalization that the inherent set a particular storage item:
+			// https://github.com/paritytech/polkadot-sdk/issues/2841#issuecomment-1876040854
+			assert!(
+				InherentIncluded::<T>::get() == true,
```

Comment:
```suggestion
				InherentIncluded::<T>::take() == true,
```

Reply:
Thank for the suggestions, I hadn't thought of this optimization